pkg/backup/remap_crd_version_action.go

```diff
@@ -89,7 +94,21 @@ func (a *RemapCRDVersionAction) Execute(item runtime.Unstructured, backup *v1.Ba
 	switch {
 	case hasSingleVersion(crd), hasNonStructuralSchema(crd), hasPreserveUnknownFields(crd):
 		log.Infof("CustomResourceDefinition %s appears to be v1beta1, fetching the v1beta version", crd.Name)
-		item, err = fetchV1beta1CRD(crd.Name, a.betaCRDClient)
+		clusterName := backup.Spec.StorageLocation
+		cluster := client.ObjectKey{
+			Namespace: clusterName,
+			Name:      clusterName,
+		}
+		restConfig, err := remote.RESTConfig(context.Background(), a.client, cluster)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		c, err := apiextensions.NewForConfig(restConfig)
+		if err != nil {
+			return nil, nil, err
+		}
+		item, err = fetchV1beta1CRD(crd.Name, c.ApiextensionsV1beta1().CustomResourceDefinitions())
```

Comment:
and c is the client to talk to the workload cluster?

Reply:
Yes, might make sense to add comment?!

Reply:
If you want (see above)

Reply:
Done. I used the word "target cluster", though, because it was already used in other comments :slightly_smiling_face:

Reply:
Thx :)